caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -410,53 +410,66 @@ func (st *ServerType) serversFromPairings(
 				return nil, fmt.Errorf("server block %v: compiling matcher sets: %v", sblock.block.Keys, err)
 			}
 
-			hosts, err := st.hostsFromServerBlockKeys(sblock.block, false)
-			if err != nil {
-				return nil, err
-			}
-
-			// tls: connection policies
-			if cpVals, ok := sblock.pile["tls.connection_policy"]; ok {
-				// tls connection policies
-				for _, cpVal := range cpVals {
-					cp := cpVal.Value.(*caddytls.ConnectionPolicy)
-
-					// make sure the policy covers all hostnames from the block
-					for _, h := range hosts {
-						if h == defaultSNI {
-							hosts = append(hosts, "")
-							cp.DefaultSNI = defaultSNI
-							break
-						}
-					}
-
-					if len(hosts) > 0 {
-						cp.MatchersRaw = caddy.ModuleMap{
-							"sni": caddyconfig.JSON(hosts, warnings), // make sure to match all hosts, not just auto-HTTPS-qualified ones
-						}
-					} else {
-						cp.DefaultSNI = defaultSNI
-						hasCatchAllTLSConnPolicy = true
-					}
-
-					srv.TLSConnPolicies = append(srv.TLSConnPolicies, cp)
-				}
-			}
+			httpHosts := make([]string, 0)
+			httpsHosts := make([]string, 0)
 
-			// exclude any hosts that were defined explicitly with
-			// "http://" in the key from automated cert management (issue #2998)
 			for _, key := range sblock.block.Keys {
 				addr, err := ParseAddress(key)
 				if err != nil {
 					return nil, err
 				}
 				addr = addr.Normalize()
-				if addr.Scheme == "http" && addr.Host != "" {
-					if srv.AutoHTTPS == nil {
-						srv.AutoHTTPS = new(caddyhttp.AutoHTTPSConfig)
+
+				//TODO: do we need to do something about this default port??
```

Comment:
I am trying to determine if this pass is for an http or https server, so probably need `app.httpPort()`

Reply:
Yeah, using the configured HTTP port is always better than hard-coding a port.

Reply:
From my tests it appears that only the scheme is needed